Vostok.Hosting/VostokHost_Extensions.cs

```diff
@@ -55,12 +55,7 @@ public static VostokHost WithSigtermCancellation([NotNull] this VostokHost vosto
         {
             AppDomain.CurrentDomain.ProcessExit += (sender, e) =>
             {
-                vostokHost.ShutdownTokenSource.Cancel();
-                
-                // NOTE: ProcessExit doesn't have flag similar to CancelKeyPress to disable process killing.
-                // NOTE: We have to wait until everything is shut down.
-                // NOTE: We can't use application state there because it is changed before disposing of environment.
-                while(!vostokHost.ApplicationClosed) {}
+                vostokHost.StopAsync().GetAwaiter().GetResult();
```

Comment:
Minor nitpick: there's a synchronous `Stop` extension that does exactly the same. Might want to use for brevity.